src/pybind/mgr/dashboard/openapi.yaml

```diff
@@ -3239,9 +3239,19 @@ paths:
           application/json:
             schema:
               properties:
+                addr:
+                  description: Network Address
+                  type: string
                 hostname:
+                  description: Hostname
                   type: string
+                labels:
+                  description: Host Labels
+                  items:
+                    type: string
+                  type: array
                 status:
+                  description: Status of the Host
```

Comment:
🙈 I hate to say this but... I see here a change in the API requiring a version increase: as long as it's an additive one, since the old call would still work, we could start with "1.1"? Only solution I can see to this here is using some kind of 'decorator'/flag for certain endpoints undertaking heavy work ('unstable'/'wip'/'beta'), so we don't have to roll up versions for every change on those.

What do others think: @nizamial09 @alfonsomthd @avanthakkar ?

Reply:
According to the [proposal](https://tracker.ceph.com/issues/40909#Versioning-semantics-and-practices),
we should increase the minor version.
We have to include in the official documentation the correspondence between Ceph versión and API versions supported.

Reply:
@alfonsomthd if you mean in the auto-generated docs (yeah). What do you think about the `@unstable` flag to override strict versioning in endpoints undergoing heavy development?

Reply:
As per our daily stand-up conversation, we should define the scope of the API endpoints stability: some endpoints are meant/likely to be consumed by automation tools/scripts (i.e. bulk operations on rgw buckets, pools, etc) and therefore they should be considered stable. Some other endpoints depending on wip features that are not meant to be consumed on an automated manner (i.e. Day-1 operations). 
Following this [endpoint stability definition for unstable](https://www.mediawiki.org/wiki/API_versioning#Unstable) I would prefer to mark them as ```Experimental```. But also we can create our own definitions.
These stability levels should also be reflected on documentation.

Reply:
@alfonsomthd @epuertat I couldn't find any doc to do this in Swagger UI. It encourages Semantic versioning like 1.0.0 or arbitrary versioning like 0.99-beta. https://swagger.io/docs/specification/api-general-info/

From the OpenAPI specifications, this is what I got. https://semver.org/spec/v2.0.0.html#doesnt-this-discourage-rapid-development-and-fast-iteration

Reply:
I checked a few discussions there ([[1]](https://github.com/OAI/OpenAPI-Specification/issues/432), [[2]](https://github.com/OAI/OpenAPI-Specification/issues/719)) and it seems that the idea has been rejected. So we have 3 possibilities:
* As @nizamial09 pointed out, by following semver, v0.y for experimental/unstable endpoints (no or minimal code changes involved),
* Add a new `experimental` `tag` (we're already using tags for theme/component grouping: RGW, RBD, etc). This involves new code in our controllers logic, and might complicate presentation in the swagger-ui (as we are sorting by tags).
* Use [OpenAPI Extensions](https://swagger.io/docs/specification/openapi-extensions/) to add `x-ceph-api-unstable: true` property. This also involves changes in the controllers logic.

Reply:
As per the discussion in the standup, will go with the semver approach for versioning.

Reply:
As agreed, let's go then with the `v0.1` for `unstable` or `experimental`.

Reply:
@epuertat @alfonsomthd While changing the version I noticed some bugs, so I created a tracker https://tracker.ceph.com/issues/50855 and I think its best to keep it seperated.